module/VuFind/src/VuFind/Route/RouteGenerator.php

```diff
@@ -113,6 +104,39 @@ public function addDynamicRoutes(& $config, $routes)
         }
     }
 
+    /**
+     * Add non tab record action & re-register all record routes to support it.
+     *
+     * @param array  $config     Configuration array to update
+     * @param string $action     Action to add
+     */
+    public function addNonTabRecordAction(& $config, $action)
+    {
+        // Explicitly use RouteGenerator:: here, so even children will
+        // use the parent's variable.
```

Comment:
I think you could use `self` instead of `RouteGenerator` and it would have the same effect...